src/routes/machine.rs

```diff
@@ -45,6 +46,22 @@ pub fn get_slots(config: ConfigData) -> Vec<String> {
     return slots;
 }
 
+#[derive(Serialize)]
+pub struct SlotStatus {
+    pub id: String,
+    pub number: i32,
+    pub stocked: bool,
+}
+pub fn get_slots(config: ConfigData) -> Vec<SlotStatus> {
+    (1..=config.slot_ids.len()).zip(config.slot_ids).map(|(n, i)| {
```

Comment:
Can't we `config.slot_ids.iter().map`? Or is there no way to map with index?
Either way, this feels very messy and we should at least have more than single-letter variable names

Reply:
`.iter().enumerate()` is what we wanted. Didn't realize that existed, oops